ivy.el

```diff
@@ -3990,7 +3990,13 @@ Skip buffers that match `ivy-ignore-buffers'."
 
 (defun ivy-switch-buffer-occur ()
   "Occur function for `ivy-switch-buffer' using `ibuffer'."
-  (ibuffer nil (buffer-name) (list (cons 'name ivy--old-re))))
+  (ibuffer
+   nil (buffer-name)
+   `((or ,@(cl-loop for cand in ivy--old-cands
+                    unless (eq (get-text-property 0 'face cand)
+                               'ivy-virtual)
+                    collect `(name . ,(format "\\_<%s\\_>"
+                                              (regexp-quote cand))))))))
```

Comment:
Nitpick: Is it really necessary to use `cl-loop` when a simpler reduction will do?
```el
   `((or ,@(cl-mapcan
            (lambda (cand)
              (unless (eq (get-text-property 0 'face cand) 'ivy-virtual)
                `((name . ,(format "\\_<%s\\_>" (regexp-quote cand))))))
            ivy--old-cands)))
```

Reply:
Since it makes a difference to you, i'll change it. But I have always found `cl-loop` easier to read then map functions that require a lambda and put the sequence at the end.

Reply:
It doesn't really make a difference to me. ;)